lib/googlestaticmap.rb

```diff
@@ -117,7 +117,7 @@ def url(protocol='http')
     attrs << ["key", @api_key] if !@api_key.nil?
     attrs << ["client", @client_id] if @api_key.nil? && !@client_id.nil? && !@private_key.nil?
     path << attrs.sort_by {|k,v| k}.collect {|attr| "#{attr[0]}=#{attr[1]}"}.join("&")
-    if @api_key.nil? && !@client_id.nil? && !@private_key.nil?
+    if (!@api_key.nil? ^ !@client_id.nil?) && !@private_key.nil?
```

Comment:
Yes, I would prefer that it be ||.  This way even if someone sets both, the signature would still be sent.  

I am going to add some better validation to ensure that the user can't set both api key and client ID.  I should have that up in master shortly.  So you make this change to use || I'll merge this in and make a new release along with my validation changes.